### PR TITLE
SCons: Move platform logo/run icon to `export` folder

### PR DESCRIPTION
**Pull request by akien-mga**
Follow-up to #75932.
Alternative to #78342.

Since these icons are only used by the export plugin, it makes sense to
move them and generate the headers there.

The whole `detect.is_active()` logic seems to be a leftover from before
times, as far back as 1.0-stable it already wasn't used for anything.

So I'm removing it and moving the export icon generation to
`platform_methods`, where it makes more sense.

### CODE REVIEW

File: methods.py [hunk 1]
```diff
@@ -259,14 +259,11 @@ def generate_version_header(module_version_string=""):
 
 
 def parse_cg_file(fname, uniforms, sizes, conditionals):
-
```
Comment (akien-mga) [inline methods.py:262]: I wonder why `black` didn't fix those until I actually modified the file locally and had it touched by my pre-commit hook.

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
I'm reworking this a bit more to move the `.svg` files themselves to the `export` folder, and clean up some legacy code.

**Comment by akien-mga** [comment]
I removed the `methods.save_active_platforms(active_platforms, active_platform_ids)` which seems to predate Godot being open source, as nothing is done with this "active platforms" concept. The methods was just used to save those export icons, so I've renamed it and moved it to `platform_methods`.

File: platform/macos/export/export_plugin.cpp [hunk 2]
```diff
@@ -30,11 +30,11 @@
 
 #include "export_plugin.h"
 
-#include "../logo_svg.gen.h"
-#include "../run_icon_svg.gen.h"
 #include "codesign.h"
 #include "lipo.h"
+#include "logo_svg.gen.h"
 #include "macho.h"
+#include "run_icon_svg.gen.h"
```
Comment (YuriSizov) [inline platform/macos/export/export_plugin.cpp:37]: Wonder if there is an argument to be made about putting generated files separately from other includes.

**Review by YuriSizov — APPROVED** [review]
Can't vouch for Meson, but changes look good.

**Pull request merged**